### PR TITLE
Add `--enable-fat` flag to gmp build

### PR DESCRIPTION
**Pull request by alopezz**
### What does this PR do?

It adds the [`--enable-fat`](https://gmplib.org/manual/Build-Options#index-Fat-binary) flag when compiling `libgmp`.

### Motivation

We've had a few support cases where the Agent was crashing with a `SIGILL` / `Illegal instruction` error. Running the Agent with `gdb` showed `libgmp` to be the culprit. Some digging lead to discovering this flag that enables runtime (as opposed to build-time) detection of CPU features, which should help with this.

### Possible Drawbacks / Trade-offs

The size of the compiled library might increase as a result.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by chouquette — APPROVED** [review]
LGTM!

**Review by chouquette — CHANGES_REQUESTED** [review]
Oh wait no this isn't a cflag, this is a configure option, this won't work

File: omnibus/config/software/gmp.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@
 
 build do
   env = with_standard_compiler_flags(with_embedded_path)
-  env["CFLAGS"] << " -fPIC"
+  # With `--enable-flat`, CPU feature detection is done at runtime, improving compatibility
+  env["CFLAGS"] << " -fPIC --enable-fat"
 
   configure "--disable-static", env: env
```
Comment (chouquette) [inline omnibus/config/software/gmp.rb:39]:
```suggestion
  configure "--disable-static --enable-fat", env: env
```
this should work better

Comment (alopezz) [reply omnibus/config/software/gmp.rb]: Indeed, good catch.

**Review by chouquette — APPROVED** [review]

**Review by brett0000FF — APPROVED** [review]

**Pull request merged**